source/javascripts/controllers/_YMLController.js.erb

```diff
@@ -2,42 +2,44 @@
 
 "use strict";
 
-angular.module("BitriseWorkflowEditor").controller("YMLController", function($scope, $rootScope, $q, appService, requestService, Progress) {
+angular.module("BitriseWorkflowEditor").controller("YMLController", function($scope, $timeout, $rootScope, $q, appService, requestService, Progress) {
 
 	var viewModel = this;
-
-	$scope.appService = appService;
+	var editor;
 	viewModel.downloadAppConfigYMLPath = requestService.mode == "website" ? requestService.appConfigYMLDownloadPath() : null;
 
-	function configureACE() {
-		var editor = ace.edit("yml-editor");
-		var session = editor.getSession();
-		var renderer = editor.renderer;
-
-		editor.setTheme('ace/theme/monokai');
-		session.setMode('ace/mode/yaml');
-
-		editor.setReadOnly(false);
-		editor.setShowPrintMargin(false);
-		editor.setShowInvisibles(false);
-		editor.$blockScrolling = Infinity;
-		editor.setOptions({
-			vScrollBarAlwaysVisible: true,
-			tabSize: 2
-		});
-		editor.setValue(appService.appConfigYML);
-		editor.focus();
-		editor.gotoLine(1);
-
-		session.setUndoManager(new ace.UndoManager());
-		session.setUseWrapMode(true);
-		session.setUseSoftTabs(true);
-
-		renderer.setShowGutter(true);
+	viewModel.configureEditor = configureEditor;
```

Comment:
No need to add these methods to view model, as they are only called from within the controller.